File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size())
            return false;
        std::unordered_map<char, int> freqS;
        for(char c : s)
            freqS[c]++;
        std::unordered_map<char, int> freqT;
        for(char c : t)
            freqT[c]++;
        for (auto it = freqS.begin(); it != freqS.end(); it++)
        {
            if (freqS[it->first] != freqT[it->first])
                return false;
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) {
        return false;
    }
    const freqS = new Map();
    const freqT = new Map();
    for (const char of s)
    {
        freqS.set(char, (freqS.get(char) || 0) + 1);
    }
    for (const char of t)
    {
        freqT.set(char, (freqT.get(char) || 0) + 1);
    }
    for (const [char, freq] of freqS) {
        if (freqT.get(char) !== freq)
        {
            return false;
        }
    }
    return true;
};